filebeat/magefile.go

```diff
@@ -96,6 +97,20 @@ func Fields() error {
 	return mage.GenerateFieldsYAML("module")
 }
 
+// GoTestUnit executes the Go unit tests.
+// Use TEST_COVERAGE=true to enable code coverage profiling.
+// Use RACE_DETECTOR=true to enable the race detector.
+func GoTestUnit(ctx context.Context) error {
```

Comment:
Could we call this just `UnitTest`? Would be great if the names of the commands would be as intuitive and rememberable as possible (I know, very subjective topic).

Reply:
I think we should have targets for `unitTest` and `integrationTest`. The unitTest target would then have dependency on `goUnitTest` and `pythonUnitTest`?

Then do the same thing for with `integrationTest` by making it depend on `goIntegrationTest` and `pythonIntegrationTest`.

Then if needed we also add `benchTest` and `loadTest`.

Reply:
SGTM. Very similar to https://github.com/elastic/beats/issues/4960

For bench and load I wonder if they should be their own target or a ENV variable. We can discuss that in a follow up.